app/Models/Post.php

```diff
@@ -218,22 +218,36 @@ public function toBlinkPayload()
         // Blink expects quantity to be a number.
         $quantity = $this->quantity === null ? 0 : $this->quantity;
 
+        // Bypass Campaign->cause accessor method so the value isn't converted to an array
+        // which Customer.io does not support.
+        $campaign_cause = (string) optional($this->signup->campaign)->getAttributes()['cause'];
+
         // Fetch Campaign Website information via GraphQL.
         $campaignWebsite = app(GraphQL::class)->getCampaignWebsiteByCampaignId($this->campaign_id);
 
+        // The associated Action for this post.
+        $action = $this->actionModel;
+
         return [
             'id' => $this->id,
             'signup_id' => $this->signup_id,
             'quantity' => $quantity,
             'why_participated' => $this->signup->why_participated,
             'campaign_id' => (string) $this->campaign_id,
             'campaign_run_id' => (string) $this->signup->campaign_run_id,
-            'campaign_title' => $campaignWebsite['title'],
-            'campaign_slug' => $campaignWebsite['slug'],
+            'campaign_title' => (string) $campaignWebsite['title'],
+            'campaign_slug' => (string) $campaignWebsite['slug'],
+            'campaign_cause' => $campaign_cause,
```

Comment:
Does this need to be cast as a string as well? If it's not set `optional()` returns `null` I think? Or is it okay if this one is `null` and gets dropped?

Reply:
I cast it as a string up top when I set the variable, do you think it'd read clearer if it were done here? 

(In general, I also wasn't completely sure that any of this even _needed_ to be cast since we shouldn't have trouble if customer.io just drops them. (Maybe just to be safe? In which case it might make sense to cast _all_ the other `null`-able fields in this payload as well?))

Reply:
OH I see it now! Jeez sorry! I think it's fine as is!

Reply:
Hmm I think that's a @ngjo question - I'm not sure if it's confusing in c.io to see fields "missing"

Reply:
As [confirmed in Slack](https://dosomething.slack.com/archives/C0YGXUE01/p1563225297003600), we've removed the extra `cast`ing for these new attributes https://github.com/DoSomething/rogue/pull/896/commits/dae3df95b11c4680d68e76d42ff08264cb9174c9